File: solutions/cpp/rna-transcription/2/rna_transcription.cpp
#include "rna_transcription.h"

#include <map>
#include <string>

using namespace std;

namespace rna_transcription {

    map<char,char> tr = {{'G', 'C'}, {'C', 'G'}, {'A', 'U'}, {'T','A'}};

   char to_rna(char nuc){
        auto search = tr.find(nuc);
        return (search != tr.end()) ? search->second : ' ';
    } 
    
   string to_rna(string dna){
       string rna="";
       for(char nuc : dna){
             rna += to_rna(nuc);
       }
       return rna;
   }

}  // namespace rna_transcription

File: solutions/julia/rna-transcription/2/rna-transcription.jl
nucleotide_dict = Dict('A'=>'U', 'T'=>'A', 'C'=>'G', 'G'=>'C') 

function is_valid(dna)
    match(r"[^ACGT]",dna) == nothing
end


function to_rna(dna)
    rna=""
    if is_valid(dna) 
        for  nuc in dna
            rna = string(rna, nucleotide_dict[nuc])
        end
    else
       throw(error("Invalid DNA input.")) 
    end
    rna
end

